pulsar-client-cpp/lib/ConsumerConfiguration.cc

```diff
@@ -74,4 +74,25 @@ void ConsumerConfiguration::setUnAckedMessagesTimeoutMs(const uint64_t milliSeco
     }
     impl_->unAckedMessagesTimeoutMs = milliSeconds;
 }
+
+bool ConsumerConfiguration::isEncryptionEnabled() const {
+    return (impl_->cryptoKeyReader != NULL);
+}
+
+const CryptoKeyReader& ConsumerConfiguration::getCryptoKeyReader() const { return *(impl_->cryptoKeyReader); }
+
+ConsumerConfiguration& ConsumerConfiguration::setCryptoKeyReader(CryptoKeyReader& cryptoKeyReader) {
```

Comment:
Instead of getting a reference and storing a pointer, we should take either a copy or a `shared_ptr` from the user. That would avoid any problem on the life cycle of the crypto reader. Otherwise the use needs to know it cannot be destructed and that will lead to unexpected segfaults.

Reply:
Switched to shared_ptr.